pandas/core/internals/blocks.py

```diff
@@ -2833,6 +2833,8 @@ def _replace_coerce(
             if convert:
                 block = [b.convert(numeric=False, copy=True) for b in block]
             return block
+        if convert:
```

Comment:
can we just move 2834 here?

Reply:
Not sure how to do that. Perhaps like that ?
```python
        if not mask.any() and not convert:
            return self
        if mask.any():
            block = super()._replace_coerce(
                to_replace=to_replace,
                value=value,
                inplace=inplace,
                regex=regex,
                convert=convert,
                mask=mask,
            )
            if convert:
                block = [b.convert(numeric=False, copy=True) for b in block]
            return block
        return [self.convert(numeric=False, copy=True)]
```
There are 4 possible senarios
1. mask.any() and convert:
return [b.convert(numeric=False, copy=True) for b in block]
2. mask.any() and not convert:
return block
3. not mask.any() and convert
return [self.convert(numeric=False, copy=True)]
4. not mask.any() and not convert
return self

Reply:
i c. this actually looks more complicated than your current change